lua/toggleterm.lua

```diff
@@ -110,11 +110,20 @@ end
 function M.handle_term_enter()
   local _, term = terms.identify()
   if term then
-    term:resize()
     close_last_window(term)
   end
 end
 
+function M.handle_term_leave()
```

Comment:
What is this function for? Presumably to close floats, but I can't see why this PR needs it

Reply:
It is mainly used to save window sizes before they are closed.

Reply:
We should only bother saving sizes if a user wants us to persist size though.

Reply:
I move closing floats here so that if a terminal is changed from a float to another direction, it is no longer automatically closed. A buffer `autocmd` was used before.

Reply:
That makes sense 👍🏿